io-engine/src/grpc/v1/test.rs

```diff
@@ -0,0 +1,172 @@
+/// RPC service for testing operations
+#[derive(Debug)]
+pub struct TestService {}
+
+impl TestService {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl Default for TestService {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[cfg(feature = "fault-injection")]
```

Comment:
hmm this is not going to work very well if we want other things in testing in addition to fault injection, example: replica wiping and perhaps others in the future?

Reply:
This block is for injection only. Any other kinds of APIs can be added outside this block, with their own feature config if they need to be enabled selectively.
There's another feature "io-engine-testing", which included "fault-injection". Another test sub-features can be added there.

Reply:
But traits have to be implemented as a whole right?

Reply:
ah, yes indeed. I'll change to per-method cfg condition.

Reply:
You could add indirection by implementing injection as separate trait or just fn impls, and then in the TestRpc trait impl call into this injection fn's, example:
```rust
#[cfg(feature = injection)]
impl Injection { fn a { } fn b() { } } 
#[cfg(feature( no( injection))]
impl Injection { fn a { Not_impl } fn b() { Not_impl } }

impl TestRcp for TestService {
  fn a() { Injection::a() }
  fn b() { Injection::b() }
}
```